internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -108,69 +108,79 @@ spec:
         export KRB5_CONFIG="${KRB5_TEMP_CONF}"
         export KRB5_TRACE=/dev/stderr
 
-        /usr/bin/kinit -V $(cat /mnt/service-account-secret/principal) -k -t /tmp/keytab
+        /usr/bin/kinit -V "$(cat /mnt/service-account-secret/principal)" -k -t /tmp/keytab
 
         set -x
+
         # check if this fbc fragment is opt-in
         echo "Fetching the image bundle from $(params.fbcFragment)..."
-        PULL_SPEC_LIST=$(opm render $(params.fbcFragment) | jq -r \
+        PULL_SPEC_LIST=$(opm render "$(params.fbcFragment)" | jq -r \
         'select(.schema == "olm.bundle") | "\(.image)" | split("@")[0]' |uniq)
         for PULL_SPEC in ${PULL_SPEC_LIST}; do
             # make sure they query is done using the internal name instead of the public
-            PULL_SPEC=$(sed  's/registry.redhat.io/registry.access.redhat.com/g' <<< $PULL_SPEC)
+            PULL_SPEC="${PULL_SPEC//registry.redhat.io/registry.access.redhat.com}"
 
             echo "Attempting to fetch from ${FETCH_URL} to check if fragment is \`fbc_opt_in==true\`..."
-            fbcOptIn+=(`isfbcOptIn ${PULL_SPEC}`)
+            IFS=" " read -r -a fbcOptIn <<< "$(isFBCOptIn "${PULL_SPEC}")"
```

Comment:
Why do we need to save an array? Isn't it enough to know that at least one pull spec was false?

Originally commented here: https://github.com/hacbs-release/app-interface-deployments/pull/231#discussion_r1865754661

You could just do something like this:
```
fbcOptIn=true
for PULL_SPEC in ${PULL_SPEC_LIST}; do
    # make sure they query is done using the internal name instead of the public
    PULL_SPEC="${PULL_SPEC//registry.redhat.io/registry.access.redhat.com}"
    echo "Attempting to fetch from ${FETCH_URL} to check if fragment is \`fbc_opt_in==true\`..."
    if [ "$(isfbcOptIn "${PULL_SPEC}")" = false ]; then
        fbcOptIn=false
        
        # This is optional - it should be enough to know that one is false, but maybe we want check all of them 
        # to print to the log - but currently it doesn't print it anyway I think.
        break 
    fi
done
mustPublishIndexImage="${fbcOptIn}"
mustSignIndexImage="${fbcOptIn}"
```

Reply:
nice. I will try this change. thanks.